source/validate_id.cpp

```diff
@@ -609,64 +609,54 @@ bool idUsage::isValid<SpvOpConstantSampler>(const spv_instruction_t* inst,
   return true;
 }
 
+bool IsBaseTypeNullable(const std::vector<uint32_t>& instruction,
```

Comment:
Missing a method blurb.
In particular, it's not immediately clear why you'd need a the usedefs.
Also, should say that you expect the instruction to be a type-defining instruction, and you're checking that the type being defined is what is nullable.